### ISSUE TITLE
create_service argument 'args' issue

### ISSUE THREAD

**Issue by frioux** [frioux/Win32-ServiceManager#3]
From Matteo Bottazzi:

I'm trying to set the 'args' argument in the create_service method in order
to pass two paths to my executable ('command' argument).
It seems to me that the '_sc_install' sub places the 'args' outside the
binpath double quotes and, for this reason, it fails to create the service.

If I execute the direct sc command from cmd, it works fine this way:
```sc create "TestService" displayname="Test Service"
binPath="\"C:\dev\WindowsService.exe\" \"C:/path_number_one\"
\"C:/path_number_two\""
```
So, if I modify your _sc_install sub from this
`qw(sc create), $_[1], qq(binpath= "$_[2]") . ($_[3] ?  " $_[3]" : '')`,
to this
`qw(sc create), $_[1], "binpath= \"$_[2]".($_[3] ?  " $_[3]" : '')."\""`,
it works correctly.

This is the service creation code:

```
    $sc->create_service (
      name        => "$ServiceName",
      display     => "EPV zParser for $Profile Profile",
      description => 'Extract data from z/OS SMF (and many other log files)
and load it in a database',
      use_perl    => 0,
      use_nssm    => 0,
      command     => $WindowsService_FullPath,
      args        => '\"'.$WindowsServiceFolder.'\" \"'.$ProfileFolder.'\"',
      depends     => [],
      start       => 'auto',
      user        => '',
      password    => '',
    )
```

Am I doing something wrong or your code needs a patch?

**Comment by frioux** [comment]
Comment from @wesQ3:

The package docs have a note on the "args" param: "XXX: do these even make
sense?" so I think it's safe to say these have never been used. I think we
should drop the option since it def doesn't work as-is. Reviewing the
(weird) syntax for sc commands with arguments[1], I think it's best to just
let the user configure the full binpath string in the "command" argument.
In his case, that would be

```
    command => q(\"C:\dev\WindowsService.exe\" \"C:/path_number_one\"
\"C:/path_number_two\")
```

A link to that SC explanation and maybe an example in the docs would go a
long way I think. A GH issue would be better to discuss anything further
for posterity etc.

What do you think?

[1]: https://stackoverflow.com/a/11084834/77782

**Comment by TeoBotta90** [comment]
I totally agree with you @wesQ3

**Comment by frioux** [comment]
Cool. If someone submits a pull request that makes setting args an error
and documents the right way to do it I'll release it!

--

**Comment by TeoBotta90** [comment]
I've created a Pull Request. First one for me on GitHub, if there's something wrong in the process or in the code just le me know and I'll fix it!

**Comment by wesQ3** [comment]
Hm, looking this over I think we could have a conflict with the `check_command` option. The exist check will fail if the user provides args in the command field. This wouldn't be a big deal except that `check_command_default` is on so anyone using args will have to set `check_command => 0`. 

Maybe this just needs further documentation and descriptive errors to let the user know `check_command` doesn't work with args.

**Comment by TeoBotta90** [comment]
You're right. Also, inside the pod there is an example of `create_service` that doesn't work until you set `check_command => 0`:
...
`command     => 'C:\code\GR\script\server.pl -p 3001',`
...
This raises 'command not found'.
Doing what you suggest is the easiest thing in my opinion, it would be ok by my side: only an improvement in the documentation.

The other possibility I see is the following:
- write in the documentation that you need to use a precise syntax for the `command` arg (use double quotes)
- change the example of `create_service` in the documentation to `command     => q(\"C:\code\GR\script\server.pl\" \"-p 3001\"),`
- change the _exists sub to `sub _exists { -e (split(/\\"/,$_[1]))[1] }`

What do you think?
Again, it is simpler what you suggested ;)

**Comment by wesQ3** [comment]
Yes I think we'll go with the simpler fix unless Frew has an objection :)

**Comment by frioux** [comment]
👍

--
